### PR TITLE
support async action well

### PR DESCRIPTION
**Pull request by zhy0216**
a small fix for async action type when manual typing

### CODE REVIEW

File: src/create-slice.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 type SliceActions<Value> = {
-  [actionName: string]: (...args: never[]) => (prev: Value) => Value;
+  [actionName: string]: (
+    ...args: never[]
+  ) => (prev: Value) => Value | Promise<Value>;
```
Comment (dai-shi) [inline src/create-slice.ts:4]: I don't think this is correct. Value should be a promise if we need async.

Would you please create a reproduction you want to support with stackblitz?

Comment (zhy0216) [reply src/create-slice.ts:4]: > I don't think this is correct. Value should be a promise if we need async.
> 
> Would you please create a reproduction you want to support with stackblitz?

basically, you cannot have sync action & async action in one slice. 
Also the value has to be Promise for async action is strange... 
by the way, the official example is broken somehow https://codesandbox.io/p/sandbox/github/zustandjs/zustand-slices/tree/main/examples/02_async?file=%2Fsrc%2Fapp.tsx

Comment (dai-shi) [reply src/create-slice.ts:4]: Sorry, it requires upcoming React 19. Let's wait for it.

Meanwhile, you can try this: 
https://twitter.com/dai_shi/status/1785504766254297436
https://stackblitz.com/edit/vitejs-vite-kxigsw?file=src%2FApp.tsx

Comment (dai-shi) [reply src/create-slice.ts:4]: This might be close to what you imagine.
https://stackblitz.com/edit/vitejs-vite-n4touv?file=src%2FApp.tsx

**Comment by dai-shi** [comment]
> async action

I'm not sure what you mean by async action,
but [this async action](https://github.com/pmndrs/zustand?tab=readme-ov-file#async-actions) is not supported for slice actions. It has to return `(prev: Value) => Value`.

However, `withActions`, it's supported.

Anyway, feedback is welcome and a reproduction or code snippet would help.

**Comment by zhy0216** [comment]
> > async action
> 
> I'm not sure what you mean by async action, but [this async action](https://github.com/pmndrs/zustand?tab=readme-ov-file#async-actions) is not supported for slice actions. It has to return `(prev: Value) => Value`.
> 
> However, `withActions`, it's supported.
> 
> Anyway, feedback is welcome and a reproduction or code snippet would help.

get it. thx.
what i am trying to do is to have `inc` & `inc1` functions in the code snippet here: https://stackblitz.com/edit/vitejs-vite-qyrkei?file=src%2FApp.tsx
Notice the type error on `inc1` function. I think my code can support this situation.

**Comment by dai-shi** [comment]
Check this. As you see, inc1 button works unless in is used. But once inc is clicked, inc1 no longer works.
https://stackblitz.com/edit/vitejs-vite-gcckgg?file=src%2FApp.tsx

That's what I mean by it's not supported.

`withActions` has a different syntax, and can do async. However, I'm not sure if this is something most people expect:
https://stackblitz.com/edit/vitejs-vite-7gfvvk?file=src%2FApp.tsx

**Comment by zhy0216** [comment]
> Check this. As you see, inc1 button works unless in is used. But once inc is clicked, inc1 no longer works. https://stackblitz.com/edit/vitejs-vite-gcckgg?file=src%2FApp.tsx
> 
> That's what I mean by it's not supported.
> 
> `withActions` has a different syntax, and can do async. However, I'm not sure if this is something most people expect: https://stackblitz.com/edit/vitejs-vite-7gfvvk?file=src%2FApp.tsx

get it! thanks

**Pull request closed without merging**